Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -61,17 +61,24 @@ void cppMain() {
 //	std::unique_ptr<sensor::ILinearPotentiometer> lin_pot(nullptr);
 //	lin_pot = std::make_unique<sensor::SLS1322>(hadc1);
 //
-	std::unique_ptr<sensor::IAccelerometer> accelerometer(nullptr);
-	accelerometer = std::make_unique<sensor::LSM6DSOX>(hi2c1);
+
+
+	auto LSM6DSOX_Accelerometer = std::make_shared<sensor::LSM6DSOX_Accelerometer>(hi2c1);
+	std::shared_ptr<sensor::IAccelerometer> accelerometer = LSM6DSOX_Accelerometer;
 	accelerometer->init();
-	static_cast<sensor::LSM6DSOX*>(accelerometer.get())->SetODR(sensor::LSM6DSOX::SensorConfiguration::ODR12_5);
```

Comment:
What was the static_cast originally for? I think I may have overheard some conversations about this around the shop.
And why was it removed?

Reply:
the static cast was because  SetODR() and SetFSR() were functions that were defined in the concrete class which made it kind of weird to be able to call these functions. This particular static cast was one of the ways that I was able to access the concrete class's unique functions.